From 159d527a5acd1e191430e380d66996c411b34785 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Date: Wed, 26 Feb 2020 15:59:37 -0800
Subject: [PATCH 0431/1921] mmc: octeontx2: Configure flags for T96 pass B0

Change-Id: I980bfde7db3fec13ae73d33964215bbf377e7a37
Signed-off-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/24021
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-by: Aaron Williams <awilliams@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/24031
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/24083
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/26926
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/mmc/host/cavium-thunderx.c | 3 +++
 drivers/mmc/host/cavium.h          | 1 +
 2 files changed, 4 insertions(+)

diff --git a/drivers/mmc/host/cavium-thunderx.c b/drivers/mmc/host/cavium-thunderx.c
index b94f1d007d1a..3ee343c0bbb7 100644
--- a/drivers/mmc/host/cavium-thunderx.c
+++ b/drivers/mmc/host/cavium-thunderx.c
@@ -235,6 +235,9 @@ static int thunder_mmc_probe(struct pci_dev *pdev,
 			host->calibrate_glitch = true;
 			host->cond_clock_glitch = true;
 			host->max_freq = MHZ_100;
+		} else if (rev == REV_ID_1) {
+			host->cond_clock_glitch = true;
+			host->max_freq = MHZ_167;
 		} else if (rev == REV_ID_2) {
 			host->tap_requires_noclk = true;
 			host->max_freq = MHZ_112_5;
diff --git a/drivers/mmc/host/cavium.h b/drivers/mmc/host/cavium.h
index 65ac8680ad2d..8bc698f38b0b 100644
--- a/drivers/mmc/host/cavium.h
+++ b/drivers/mmc/host/cavium.h
@@ -36,6 +36,7 @@
 
 /* Chip revision Id */
 #define REV_ID_0 0
+#define REV_ID_1 1
 #define REV_ID_2 2
 
 #define KHZ_400 (400000)
-- 
2.31.1

